corehq/apps/app_manager/app_schemas/case_properties.py

```diff
@@ -155,32 +429,39 @@ def get_parent_type_map(self, case_types, allow_multiple_parents=False):
                         )
                     parent_map[case_type][relationship] = types[0]
 
+        if case_types is not None:
+            return {case_type: rel_map for case_type, rel_map in parent_map.items()
+                    if case_type in case_types}
         return parent_map
 
-    def get_case_property_map(self, case_types,
-                              include_shared_properties=True,
-                              include_parent_properties=True):
+    def get_case_property_map(self, case_types):
+        """
+        Same data as self.get_properties_by_case_type, filtered and sorted.
+
+        Include only `case_type`s mentioned in `case_types`,
+        and put properties in a sorted list.
+
+        :param case_types: case_types to filter on
+
+        :return: {case_type: [property]}, with the property list sorted alphabetically.
+        """
         case_types = sorted(case_types)
         return {
-            case_type: sorted(self.get_properties(
-                case_type,
-                include_shared_properties=include_shared_properties,
-                include_parent_properties=include_parent_properties,
-            ))
+            case_type: sorted(self.get_properties(case_type))
             for case_type in case_types
         }
 
 
-def get_case_properties(app, case_types, defaults=(),
-                        include_shared_properties=True,
-                        include_parent_properties=True):
+def get_parent_type_map(app):
+    builder = ParentCasePropertyBuilder(app)
+    return builder.get_parent_type_map(app.get_case_types())
```

Comment:
What do you think of passing `allow_multiple_parents=True` here? or maybe at least expose that parameter so it can be passed if needed. Otherwise this will return incomplete and I think even nondeterministic results for some apps.

Reply:
Okay, I changed the name of this arg, made the good value the default one, and made the one place that uses the bad value pass that in explicitly. Instead of `allow_multiple_parents=False` (which was the default) you now have to pass in `if_multiple_parents_arbitrarily_pick_one=True` explicitly.